File: demo/src/main/java/com/fengshuisystem/demo/entity/PostCategory.java
package com.fengshuisystem.demo.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.Nationalized;

import java.util.LinkedHashSet;
import java.util.Set;

@Getter
@Setter
@Entity
public class PostCategory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "post_category_id", nullable = false)
    private Integer id;

    @Size(max = 255)
    @NotNull
    @Nationalized
    @Column(name = "post_category_name", nullable = false)
    private String postCategoryName;

    @OneToMany(mappedBy = "postCategory", cascade = {CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REFRESH, CascadeType.DETACH})
    private Set<Post> posts = new LinkedHashSet<>();

}

File: src/models/PostCategory.ts
import { Status } from './enums/Status';
import { Post } from './Post';

export class PostCategory {
    postCategoryId: number;
    postCategoryName: string;
    status: Status;
    posts: Set<Post>;

    constructor() {
        this.postCategoryId = 0;
        this.postCategoryName = '';
        this.status = Status.INACTIVE;
        this.posts = new Set<Post>();
    }
}
